File: c/Vetores/dados_pessoas.c
#include <stdio.h>

int main() {

    int n;
    double alturas[100];
    char generos[100];

    double maiorAltura;
    double menorAltura;
    double somaAlturaMulheres = 0;
    double mediaMulheres;

    int quantidadeMulheres = 0;
    int quantidadeHomens = 0;

    printf("Quantas pessoas voce vai digitar? ");
    scanf("%d", &n);

    for (int i = 0; i < n; i++) {

        printf("\nDados da %da pessoa:\n", i + 1);

        printf("Altura: ");
        scanf("%lf", &alturas[i]);

        printf("Genero (M/F): ");
        scanf(" %c", &generos[i]);

        // Inicializa maior e menor usando a primeira pessoa
        if (i == 0) {
            maiorAltura = alturas[i];
            menorAltura = alturas[i];
        }

        if (alturas[i] > maiorAltura) {
            maiorAltura = alturas[i];
        }

        if (alturas[i] < menorAltura) {
            menorAltura = alturas[i];
        }

        if (generos[i] == 'F') {
            somaAlturaMulheres += alturas[i];
            quantidadeMulheres++;
        }
        else if (generos[i] == 'M') {
            quantidadeHomens++;
        }
    }

    printf("\nMAIOR ALTURA = %.2lf\n", maiorAltura);
    printf("MENOR ALTURA = %.2lf\n", menorAltura);

    if (quantidadeMulheres > 0) {
        mediaMulheres = somaAlturaMulheres / quantidadeMulheres;
        printf("MEDIA DAS ALTURAS DAS MULHERES = %.2lf\n", mediaMulheres);
    }
    else {
        printf("NAO HA MULHERES NO GRUPO\n");
    }

    printf("NUMERO DE HOMENS = %d\n", quantidadeHomens);

    return 0;
}

File: java/Vetores/DadosPessoas.java
import java.util.Locale;
import java.util.Scanner;

public class DadosPessoas {
    public static void main(String[] args) {

        Locale.setDefault(Locale.US);
        Scanner sc = new Scanner(System.in);

        int N;
        int homens = 0;
        int mulheres = 0;
        double somaAlturaMulheres = 0;

        double maiorAltura = 0;
        double menorAltura = 0;

        System.out.print("Quantas pessoas serao digitadas: ");
        N = sc.nextInt();

        for (int i = 0; i < N; i++) {

            System.out.print("Digite a altura: ");
            double altura = sc.nextDouble();

            System.out.print("Digite o genero (M/F): ");
            char genero = sc.next().charAt(0);

            if (i == 0) {
                maiorAltura = altura;
                menorAltura = altura;
            }

            if (altura > maiorAltura) {
                maiorAltura = altura;
            }

            if (altura < menorAltura) {
                menorAltura = altura;
            }

            if (genero == 'F') {
                somaAlturaMulheres += altura;
                mulheres++;
            } 
            else if (genero == 'M') {
                homens++;
            }
        }

        double mediaMulheres = somaAlturaMulheres / mulheres;

        System.out.printf("Menor altura = %.2f%n", menorAltura);
        System.out.printf("Maior altura = %.2f%n", maiorAltura);
        System.out.printf("Media das alturas das mulheres = %.2f%n", mediaMulheres);
        System.out.println("Numero de homens = " + homens);

        sc.close();
    }
}